src/Apps/Artist/Components/ArtistHeader.tsx

```diff
@@ -36,27 +36,14 @@ export class ArtistHeader extends Component<Props> {
       <ContextConsumer>
         {({ mediator, user }) => {
           return (
-            <Responsive>
-              {({ xs }) => {
-                if (xs) {
-                  return (
-                    <SmallArtistHeader
-                      mediator={mediator}
-                      user={user}
-                      {...props}
-                    />
-                  )
-                } else {
-                  return (
-                    <LargeArtistHeader
-                      mediator={mediator}
-                      user={user}
-                      {...props}
-                    />
-                  )
-                }
-              }}
-            </Responsive>
+            <>
+              <breakpoints.xs>
+                <SmallArtistHeader mediator={mediator} user={user} {...props} />
+              </breakpoints.xs>
+              <breakpoints.xs not>
+                <LargeArtistHeader mediator={mediator} user={user} {...props} />
+              </breakpoints.xs>
```

Comment:
```jsx
<breakpoints.xs>small</breakpoints.xs>
<breakpoints.xs not>everything else</breakpoints.xs>
```

I think we should continue iterating on feel of this API as its a bit unintuitive and error prone and not to conducive to large-scale refactors. Why remove `<breakpoints.else>`? That seemed pretty clear and maps to a good mental model at a glance. A factory function like you alluded to

```js
const { xs, rest } = makeBreakpoints([breakpoints.xs]) 

<xs>
  ...
</xs>
<rest>
  ...
</rest>
```
seems flexible while also clear.

Reply:
`breakpoints.else` was too smart. It required to be directly nested under the `Responsive` component, which seemed both hard to explain and a bit much. (Imagine needing to wrap multiple levels in a render function with the Responsive component.)

But I like your example of upfront specifying what breakpoints will be used and thus being able to create the `else` case upfront. Care to sketch that out further?

Reply:
Ah, updated comment with a new renderProps example which i think would be the most flexible since those values could be passed at runtime and / or expanded to other kinds of media queries.

Reply:
I think we should avoid render props where possible, as I find them to make readability of large pieces of markup harder.